core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationService.java

```diff
@@ -121,35 +121,45 @@ public FlowIdentifier removeInstances(Stack stack, Collection<String> instanceId
-                updateNodeCountValidator.validateScalabilityOfInstanceGroup(stack, instanceGroupName, scalingAdjustment);
-                updateNodeCountValidator.validateScalingAdjustment(instanceGroupName, scalingAdjustment, stack);
-            }
-        }
-        if (scalingStrategy != null) {
-            // TODO: CB-14929: CB-15154 stopstart can only support a single hostGroup being specified.
-            // TODO: CB-14929: Should this not be putting the stack into a UpdateInProgress state?, and maybe run inside a transaction? (similar to the upscale)
-            if (instanceIdsByHostgroupMap.size() > 1) {
-                throw new BadRequestException("Downscale via Instance Stop cannot process more than one host group");
+        try {
+            return transactionService.required(() -> {
+                Map<String, Set<Long>> instanceIdsByHostgroupMap = new HashMap<>();
+                for (String instanceId : instanceIds) {
+                    // TODO CB-14929: CB-15154 An invalid instanceId should not necessarily fail a downscale-stop operation?
+                    InstanceMetaData metaData = updateNodeCountValidator.validateInstanceForDownscale(instanceId, stack);
+                    instanceIdsByHostgroupMap.computeIfAbsent(metaData.getInstanceGroupName(), s -> new LinkedHashSet<>()).add(metaData.getPrivateId());
+                }
+                updateNodeCountValidator.validateServiceRoles(stack, instanceIdsByHostgroupMap.entrySet()
+                        .stream()
+                        .collect(Collectors.toMap(Entry::getKey, e -> e.getValue().size() * -1)));
+                LOGGER.info("ZZZ: removeInstances: instanceIds/hostnames received: {}", instanceIds);
+                LOGGER.info("ZZZ: removeInstances: computedInstanceIds: {}", instanceIdsByHostgroupMap);
+                // TODO CB-14929: Validations for start-stop
+                if (!forced) {
+                    for (Entry<String, Set<Long>> entry : instanceIdsByHostgroupMap.entrySet()) {
+                        String instanceGroupName = entry.getKey();
+                        int scalingAdjustment = entry.getValue().size() * -1;
+                        updateNodeCountValidator.validateScalabilityOfInstanceGroup(stack, instanceGroupName, scalingAdjustment);
+                        updateNodeCountValidator.validateScalingAdjustment(instanceGroupName, scalingAdjustment, stack);
+                    }
+                }
+                if (scalingStrategy != null) {
+                    // TODO: CB-14929: CB-15154 stopstart can only support a single hostGroup being specified.
+                    if (instanceIdsByHostgroupMap.size() > 1) {
+                        throw new BadRequestException("Downscale via Instance Stop cannot process more than one host group");
+                    }
+                    stackUpdater.updateStackStatus(stack.getId(), DetailedStackStatus.DOWNSCALE_BY_STOP_REQUESTED,
```

Comment:
This would make the change for the DOWNSCALE_VIA_STOP, but not for the regular DOWNSCALE - i.e. the DB level change.

Reply:
I see DOWNSCALE_REQUESTED is updated in a different flow. Not sure why scalingStrategy != null check is even performed here. I do not see a similar check during upscale flow by start.

Reply:
sclaingSTrategy != null -> This is because removeInstances is used for both regular downscales as well as stopstart-downscales.

Reply:
I see, then I do not need to update here, because it is already updated in another method,

```
    public FlowIdentifier updateNodeCount(Stack stack, InstanceGroupAdjustmentV4Request instanceGroupAdjustmentJson, boolean withClusterEvent) {
        environmentService.checkEnvironmentStatus(stack, EnvironmentStatus.upscalable());
        try {
            return transactionService.required(() -> {
                Stack stackWithLists = stackService.getByIdWithLists(stack.getId());
                updateNodeCountValidator.validateServiceRoles(stackWithLists, instanceGroupAdjustmentJson);
                updateNodeCountValidator.validateStackStatus(stackWithLists);
                updateNodeCountValidator.validateInstanceGroup(stackWithLists, instanceGroupAdjustmentJson.getInstanceGroup());
                updateNodeCountValidator.validateScalabilityOfInstanceGroup(stackWithLists, instanceGroupAdjustmentJson);
                updateNodeCountValidator.validateScalingAdjustment(instanceGroupAdjustmentJson, stackWithLists);
                updateNodeCountValidator.validateInstanceStatuses(stackWithLists, instanceGroupAdjustmentJson);
                if (withClusterEvent) {
                    updateNodeCountValidator.validateClusterStatus(stackWithLists);
                    updateNodeCountValidator.validateHostGroupIsPresent(instanceGroupAdjustmentJson, stackWithLists);
                    updateNodeCountValidator.validataHostMetadataStatuses(stackWithLists, instanceGroupAdjustmentJson);
                }
                if (instanceGroupAdjustmentJson.getScalingAdjustment() > 0) {
                    stackUpdater.updateStackStatus(stackWithLists.getId(), DetailedStackStatus.UPSCALE_REQUESTED,
                            "Requested node count for upscaling: " + instanceGroupAdjustmentJson.getScalingAdjustment());
                    return flowManager.triggerStackUpscale(
                            stackWithLists.getId(),
                            instanceGroupAdjustmentJson,
                            withClusterEvent);
                } else {
                    stackUpdater.updateStackStatus(stackWithLists.getId(), DetailedStackStatus.DOWNSCALE_REQUESTED,
                            "Requested node count for downscaling: " + abs(instanceGroupAdjustmentJson.getScalingAdjustment()));
                    return flowManager.triggerStackDownscale(stackWithLists.getId(), instanceGroupAdjustmentJson);
                }
            });
        } catch (TransactionExecutionException e) {
            if (e.getCause() instanceof BadRequestException) {
                throw e.getCause();
            }
            throw new TransactionRuntimeExecutionException(e);
        }
    }
```